apps/test-suite/tests/Battery.js

```diff
@@ -0,0 +1,104 @@
+                    wasRejected = true;
+                }
+                expect(batteryLevel).toBeLessThanOrEqual(1);
+                expect(batteryLevel).toBeGreaterThan(0);
+                expect(wasRejected).toBe(false);
+            });
+            it('Gets battery state', async () => {
+                let batteryState;
+                let wasRejected = false;
+                try {
+                    batteryState = await Battery.getBatteryStateAsync();
+                } catch (error) {
+                    wasRejected = true;
+                }
+                expect(batteryState).toBeDefined();
+                expect(batteryState).toEqual(jasmine.any(String));
+                expect(wasRejected).toBe(false);
+            });
+            it('Gets low power mode', async () => {
+                let lowPowerMode;
+                let wasRejected = false;
+                try {
+                    lowPowerMode = await Battery.getLowPowerModeStatusAsync();
+                } catch (error) {
+                    wasRejected = true;
+                }
+                expect(lowPowerMode).toBeDefined();
+                expect(lowPowerMode).toEqual(jasmine.any(String));
+                expect(wasRejected).toBe(false);
+            });
+            it('Gets power state', async () => {
+                let powerState;
+                let wasRejected = false;
+                try {
+                    powerState = await Battery.getPowerStateAsync();
+                } catch (error) {
+                    wasRejected = true;
+                }
+                expect(powerState).toBeDefined();
+                expect(wasRejected).toBe(false);
+            });
+        });
+
+        describe('Event listeners', () => {
+            let _subscriptionBatteryLevel, _subscriptionBatteryState, _subscriptionPowerMode;
+
+            //event listeners do register
+            it('Registers event listeners', () => {
+                let hasError = false;
+                try {
+                    _subscriptionPowerMode = Battery.watchPowerModeChange(({ lowPowerMode }) => {
+                        console.log('powerMode changed!', lowPowerMode);
+                    });
+                    _subscriptionBatteryState = Battery.watchBatteryStateChange(({ batteryState }) => {
+                        console.log('batteryState changed!', batteryState);
+                    });
+                    _subscriptionBatteryLevel = Battery.watchBatteryLevelChange(({ batteryLevel }) => {
+                        console.log('batteryLevel changed!', batteryLevel);
+                    });
+                } catch(e) {
+                    console.log(e);
+                    hasError = true;
+                }
+                expect(hasError).toBe(false);
+                expect(_subscriptionBatteryLevel).toBeDefined();
+                expect(_subscriptionBatteryState).toBeDefined();
+                expect(_subscriptionPowerMode).toBeDefined();
```

Comment:
to make those two tests independent of each other, clean up after each one 🙂 otherwise failure of the first one would make the second one pass even though it could be faulty